File: services/auth/service-auth/src/main/java/com/mlacker/micros/auth/service/token/TokenFactory.java
package com.mlacker.micros.auth.service.token;

import com.auth0.jwt.JWT;
import com.auth0.jwt.algorithms.Algorithm;
import com.mlacker.micros.auth.domain.user.Account;
import com.mlacker.micros.config.properties.TokenProperties;
import java.sql.Date;
import java.time.LocalDateTime;
import java.time.ZoneId;
import org.springframework.stereotype.Component;

@Component
public class TokenFactory {

    private final TokenProperties properties;

    public TokenFactory(TokenProperties properties) {
        this.properties = properties;
    }

    public String createToken(Account account) {
        if (account == null) {
            throw new IllegalArgumentException("Cannot create token without account");
        }

        LocalDateTime currentTime = LocalDateTime.now();

        return JWT.create()
                .withIssuer(properties.getIssuer())
                .withIssuedAt(Date.from(currentTime.atZone(ZoneId.systemDefault()).toInstant()))
                .withExpiresAt(Date.from(currentTime.plusMinutes(properties.getExpires())
                        .atZone(ZoneId.systemDefault()).toInstant()))
                .withSubject(String.valueOf(account.getId()))
                .withClaim("name", account.getName())
                .withClaim("app", "1138238647")
                .sign(Algorithm.HMAC256(properties.getSigningKey()));
    }
}

File: services/service-auth/service-auth/src/main/kotlin/com/mlacker/micros/auth/domain/user/TokenFactory.kt
package com.mlacker.micros.auth.domain.user

import com.auth0.jwt.JWT
import com.auth0.jwt.algorithms.Algorithm
import com.mlacker.micros.config.properties.TokenProperties
import org.springframework.stereotype.Component
import java.sql.Date
import java.time.LocalDateTime
import java.time.ZoneId

@Component
class TokenFactory(private val properties: TokenProperties) {

    fun createToken(account: Account): String {
        val currentTime = LocalDateTime.now()

        return JWT.create()
            .withIssuer(properties.issuer)
            .withIssuedAt(Date.from(currentTime.atZone(ZoneId.systemDefault()).toInstant()))
            .withExpiresAt(
                Date.from(
                    currentTime.plusMinutes(properties.expires.toLong())
                        .atZone(ZoneId.systemDefault()).toInstant()
                )
            )
            .withSubject(account.id.toString())
            .withClaim("name", account.name)
            .withClaim("app", "1138238647")
            .sign(Algorithm.HMAC256(properties.signingKey))
    }
}